js/phyx2jsonld.js

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env node
+
+/*
+ * phyx2jsonld.js is a script that provides an interface to phyx.js that allows
+ * users to convert PHYX files to JSON-LD at the command line.
+ *
+ *  ./phyx2jsonld.js input.json [-o output.jsonld]
+ *
+ * Where:
+ *  - input.json is the PHYX file to be converted.
+ *  - output.jsonld is the JSON-LD file to be produced. If no output filename is
+ *    given, the output will be written to STDOUT.
+ */
+
+/* Load Javascript libraries */
+// We use 'fs' for input/output.
+const fs = require('fs');
+
+// Load yargs, a command line argument
+const yargs = require('yargs');
+
+// Get stdin, which helps read input from STDIN.
+const getStdin = require('get-stdin');
+
+/*
+ * phyx.js uses some code (in particular through phylotree.js) that expects certain
+ * Javascript libraries to be loaded via the browser using <script>. To replicate
+ * this in Node, we load them and add them to the global object.
+ */
+
+// Load moment as a global variable so it can be accessed by phyx.js.
+global.moment = require('moment');
+
+// Load jQuery.extend as a global variable so it can be accessed by phyx.js.
+global.jQuery = {};
+global.jQuery.extend = require('extend');
+
+// Load d3 as a global variable so it can be accessed by both phylotree.js (which
+// needs to add additional objects to it) and phyx.js (which needs to call it).
+global.d3 = require('d3');
+
+// phylotree.js does not export functions itself, but adds them to global.d3.layout.
+// So we set up a global.d3.layout object for them to be added to, and then we include
+// phylotree.js ourselves.
+if (!Object.prototype.hasOwnProperty.call(global.d3, 'layout')) {
+  global.d3.layout = {};
+}
+require('../lib/phylotree.js/phylotree.js');
```

Comment:
I guess it's a little weird to see this loaded with absolute path rather than just be a part of the dependencies that get installed through `package.json` and npm. Is there a reason it is not the standard way?

Reply:
Nope, there isn't! Before we had an npm setup, I just included the JavaScript and CSS files directly in the repository. I've opened an issue to move us to using the npm-published version of Phyloref.js (#121).